pkg/segment/results/mresults/metricresults.go

```diff
@@ -375,6 +375,36 @@ func (r *MetricsResult) GetResultsPromQl(mQuery *structs.MetricsQuery, pqlQueryt
 	return retVal, nil
 }
 
+func (res *MetricsResult) GetMetricTagsResultSet(mQuery *structs.MetricsQuery) ([]string, []string, error) {
+	if res.State != SERIES_READING {
+		return nil, nil, errors.New("results is not in Series Reading state")
+	}
+
+	tagKeysMap := make(map[string]bool)
+	uniqueTagKeys := make([]string, 0)
+	for _, tag := range mQuery.TagsFilters {
+		if _, ok := tagKeysMap[tag.TagKey]; !ok {
+			tagKeysMap[tag.TagKey] = true
+			uniqueTagKeys = append(uniqueTagKeys, tag.TagKey)
+		}
+	}
+
+	tagKeyValueSet := make([]string, 0)
+
+	for _, series := range res.AllSeries {
+		tagValues := strings.Split(series.grpID.String(), tsidtracker.TAG_VALUE_DELIMITER_STR)
+		if len(uniqueTagKeys) != len(tagValues)-1 {
+			err := fmt.Errorf("GetResults: the length of tag key and tag value pair must match. UniqueTagKeys length: %v,  TagValues Length: %v", len(uniqueTagKeys), len(tagValues)-1)
+			return nil, nil, err
+		}
+		for index, val := range tagValues[:len(tagValues)-1] {
+			tagKeyValueSet = append(tagKeyValueSet, fmt.Sprintf("%s:%s", uniqueTagKeys[index], val))
```

Comment:
- Should I return here? Or Log the Error and continue to process the next series?

- Also, wouldn't there be a case or scenario where the Unique Tag Keys for the Metric may not match with the Tag Values?
   - I meant to say that for a Metric (`Metric_1`), some values may include two tag keys and values. For example: `method: post, respCode: 200`. But some may only include `method: post` with no `respCode`. In this scenario , the `uniqueTagKeysMap` will have both `method` and `respCode`, but for some series, there will not be value for `respCode`.
   - Or is there a general assumption that the above scenario won't happen. Meaning for a given Metric, the tag set will remain constant?